test/e2e/apps/replica_set.go

```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
 	framework.ExpectEqual(*(rs.Spec.Replicas), int32(4), "replicaset should have 4 replicas")
 
 }
+
+// ReplicaSet Replace and Patch tests
+func testRSLifeCycle(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+	zero := int64(0)
+
+	// Create webserver pods.
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+	}
+
+	rsName := "test-rs"
+	replicas := int32(1)
+	rsPatchReplicas := int32(3)
+	rsPatchImage := imageutils.GetE2EImage(imageutils.Pause)
+
+	// Create a ReplicaSet
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	_, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Verify that the required pods have come up.
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "error in waiting for pods to come up: %s", err)
+
+	// Scale the ReplicaSet
+	ginkgo.By(fmt.Sprintf("Scaling up %q replicaset ", rsName))
+	_, err = e2ereplicaset.UpdateReplicaSetWithRetries(c, ns, rsName, func(update *appsv1.ReplicaSet) {
+		x := int32(2)
+		update.Spec.Replicas = &x
+	})
+	framework.ExpectNoError(err)
+
+	// Patch the PeplicaSet
+	ginkgo.By("patching the ReplicaSet")
+	rsPatch, err := json.Marshal(map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": map[string]string{"test-rs": "patched"},
+		},
+		"spec": map[string]interface{}{
+			"replicas": rsPatchReplicas,
+			"template": map[string]interface{}{
+				"spec": map[string]interface{}{
+					"TerminationGracePeriodSeconds": &zero,
+					"containers": [1]map[string]interface{}{{
+						"name":    rsName,
+						"image":   rsPatchImage,
+						"command": []string{"/bin/sleep", "100000"},
+					}},
+				},
+			},
+		},
+	})
+	framework.ExpectNoError(err, "failed to Marshal Deployment JSON patch")
```

Comment:
s/Deployment/ReplicaSet ?

Reply:
Good catch, thank you.
Updated to ReplicaSet